editor/src/components/context-menu-wrapper.tsx

```diff
@@ -18,142 +27,125 @@ import type { WindowPoint } from '../core/shared/math-utils'
 import { windowPoint } from '../core/shared/math-utils'
 import { BodyMenuOpenClass } from '../core/shared/utils'
 
-export interface ContextMenuWrapperProps<T> {
-  id: string
-  items: Array<ContextMenuItem<T>>
-  dispatch?: EditorDispatch
-  data: T
-  renderTag?: string
+interface Submenu<T> {
+  items: Item<T>[]
+  label: string | ReactNode
+  type: 'submenu'
+}
+
+interface SimpleItem<T> {
+  item: Item<T>
+  type: 'simple'
+}
+
+type MenuItem<T = null> = Submenu<T> | SimpleItem<T> | null
+
+export type Item<T = null> = ContextMenuItem<T> | null
```

Comment:
Low key, I don't love how I've defaulted to null here, but it works.

Haven't dealt much with generics, used to preferring to have explicit typing — so open to any suggestions. Particularly I hadn't encountered component's having types to them, ie. `export class MomentumContextMenu<T> extends ReactComponent<ContextMenuProps<T>>` where this first `T` is coming from and how to seamlessly translate to functional component refactor.